src/cmd-buildextend-azure

```diff
@@ -0,0 +1,188 @@
+            " * size is %s",
+            self._found_files[tmp_img_azure_vhd]["size"])
+        # ---
+
+
+def cli():
+    """
+    Parse args and dispatch
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        '--auth', help='Path to Azure auth file',
+        required=True, default=os.environ.get("AZURE_AUTH"))
+    parser.add_argument('--build', help='Build ID', required=True)
+    parser.add_argument(
+        '--container', help='Storage location to write to',
+        required=True, default=os.environ.get("AZURE_CONTAINER"))
+    parser.add_argument(
+        '--force', help='Replace existing images and upload',
+        action='store_true',
+        default=bool(os.environ.get('AZURE_FORCE', False)))
+    parser.add_argument(
+        '--location', help='Azure location (default westus)',
+        required=false)
+    parser.add_argument(
+        '--profile', help='Path to Azure profile',
+        required=True, default=os.environ.get('AZURE_PROFILE'))
+    parser.add_argument(
+        '--resource-group', help='Resource group', required=True,
+        default=os.environ.get('AZURE_RESOURCE_GROUP'))
+    parser.add_argument(
+        '--storage-account', help='Storage account', required=True,
+        default=os.environ.get('AZURE_STORAGE_ACCOUNT'))
+    args = parser.parse_args()
+
+    # Argument checks for environment strings that are required
+    arg_exp_str = "parameter '--{}' or envVar '{}' must be defined"
+    if args.auth is None:
+        raise Exception(arg_exp_str.format('auth', 'AZURE_AUTH'))
+    if args.container is None:
+        raise Exception(arg_exp_str.format('container', 'AZURE_CONTAINER'))
+    if args.profile is None:
+        raise Exception(arg_exp_str.format('profile', 'AZURE_PROFILE'))
+    if args.resouce_group is None:
+        raise Exception(arg_exp_str.format('resource_group', 'AZURE_RESOURCE_GROUP'))
+    if args.storage_account is None:
+        raise Exception(arg_exp_str.format('storage_account', 'AZURE_STORAGE_ACCOUNT'))
+
+    # Identify the builds
+    build = Build(
+        os.path.join('builds', args.build),
+        args.build,
+        workdir=os.path.abspath('tmp/buildpost-azure'))
+
+    if 'azure' in build.meta['images'] and not args.force:
+        print('Azure image already built; use --force to rebuild')
+        raise SystemExit()
+
+    build.build_artifacts(cli_args=args)
+    run_ore(args, build)
+
+
+def run_ore(args, build):
+    """
+    Execute ore to upload the vhd image in blob format
+    See:
+      - https://github.com/coreos/mantle/#azure
+      - https://docs.microsoft.com/en-us/azure/storage/blobs/storage-blobs-introduction
+    TODO: Move to using an _Upload subclass.
+    :param args: The command line arguments
+    :type args: argparse.Namespace
+    :param build: Build instance to use
+    :type build: Build
+    """
+    blob_url = f'https://{args.storage_account}.blob.core.windows.net/{args.container}/{build.azure_vhd_name}'
```

Comment:
This doesn't look used anywhere.

Reply:
Right, we should probably be adding that to the json?